src/snotify/snotify.service.ts

```diff
@@ -22,7 +24,10 @@ export class SnotifyService {
   readonly toastDeleted = new Subject<number>();
   private notifications: SnotifyToast[] = [];
 
-  constructor(@Inject('SnotifyToastConfig') public config: SnotifyDefaults) {
+  public config: ProcessedSnotifyDefaults;
+
+  constructor(@Inject('SnotifyToastConfig') config: SnotifyDefaults) {
+    this.config = mergeDeep(ToastDefaults, config) as ProcessedSnotifyDefaults;
```

Comment:
As far as I can tell, before this update the user provided `SnotifyDefaults` were used "as is" for the `config` property. I think this would open up a bug if the user provided an incomplete `SnotifyDefaults` object. Here's I'm merging the user provided `config` with `ToastDefaults` to produce the `ProcessedSnotifyDefaults`.